### PR TITLE
Add dag-scope tests

### PR DESCRIPTION
**Pull request by guanzo**
TODO:

* [x] Add a fixture that supports comprehensive car-scope tests.
* [x] Rename `car-scope` to `dag-scope` once lassie supports it.

How I generated the fixture
```sh
#!/bin/bash

# set kubo config "Internal.UnixFSShardingSizeThreshold": "10B"

mkdir -p dir/sub1
echo "abc123" > dir/sub1/hello.txt
mkdir -p dir/sub2
echo "xyz456" > dir/sub2/hello.txt

ROOT_CID=$(ipfs add -Qr --chunker=size-5 dir)

ipfs dag export ${ROOT_CID} > ./t0118-ipip-402-scope.car
```

The point of 2 subdirectories is that if the request is for one of them (`/sub1/hello.txt`), than the other (`/sub2`) should not appear in the CAR.

### CODE REVIEW

File: tests/t0118_gateway_car_test.go [hunk 1]
```diff
@@ -44,10 +44,87 @@ func TestGatewayCar(t *testing.T) {
...
+				),
+		},
+		{
+			Name: "GET with ?format=car&car-scope=block params returns expected blocks",
+			Hint: `
+				car-scope=block should return a CAR file with only the root block and a
+				block for each optional path component.
+			`,
+			Request: Request().
+				Path("ipfs/%s/subdir/ascii.txt", fixture.MustGetCid()).
```
Comment (laurentsenta) [inline tests/t0118_gateway_car_test.go:64]:
```suggestion
				Path("ipfs/{{cid}}/subdir/ascii.txt", fixture.MustGetCid()).
```

Note we introduced templates to work around the collision between `%verbs` and HTML escape codes, etc.
https://github.com/ipfs/gateway-conformance/pull/41

You'll have to rename these formatting.

**Comment by laurentsenta** [comment]
Adding related contributors for review, the code looks fine to me;
Could you rebase over the latest main? It looks like it shows the car tooling changes that are already in main.

**Review by laurentsenta — APPROVED** [review]

**Comment by guanzo** [comment]
Had trouble doing a clean rebase so force pushed instead 😅

**Comment by galargh** [comment]
The tests haven't been running on this PR because we had a wrong trigger configured for the job. This is being fixed in https://github.com/ipfs/gateway-conformance/pull/60

In the meantime, I pushed https://github.com/ipfs/gateway-conformance/tree/guanzo-main so that we get CI runs for the HEAD of this PR. Unfortunately, it looks like the newly added tests aren't passing when running against kubo-master. See https://github.com/ipfs/gateway-conformance/actions/runs/5119205028/attempts/1#summary-13852714693

**Comment by galargh** [comment]
We can skip the some expects like this: https://github.com/ipfs/gateway-conformance/blob/26480708f3658ade912382cb2af82a0b40d3db8e/.github/workflows/test.yml#L65

**Comment by hacdias** [comment]
I'm closing this and we'll continue on https://github.com/ipfs/gateway-conformance/pull/56. Thanks @guanzo for your contribution on this. Changes are included in #56.

**Pull request closed without merging**